js/lib/ledgerExportUtil.js

```diff
@@ -0,0 +1,285 @@
+ * //   /.../
+ * // ]
+ *
+ * @param {Object[]} transactions - array of transactions
+ * @param {string[]=} viewingIds - OPTIONAL array/string with one or more viewingIds to filter transactions by (if empty, uses all tx)
+ * @param (boolean=) addTotalRow - OPTIONAL boolean indicating whether to add a TOTALS row (defaults false)
+ **/
+let getTransactionCSVRows = function (transactions, viewingIds, addTotalRow) {
+  let txContribData = getPublisherVoteData(transactions, viewingIds)
+  var publishers = (underscore.keys(txContribData) || [])
+
+  // sort publishers alphabetically
+  // TODO: take locale argument and pass to localeCompare below
+  publishers = publishers.sort(function (a, b) {
+    return (a && typeof a === 'string' ? a : '').localeCompare(b && typeof b === 'string' ? b : '')
+  })
+
+  var currency = txContribData[publishers[0]].contribution.currency
+  var headerRow = ['Publisher', 'Votes', 'Fraction', 'BTC', currency].join(',')
+
+  var totalsRow = {
+    label: 'TOTAL',
+    votes: 0,
+    fraction: 0,
+    btc: 0,
+    fiat: 0
+  }
+
+  var rows = [headerRow]
+
+  rows = rows.concat(publishers.map(function (pub) {
+    var pubRow = txContribData[pub]
+
+    let rowBTC = pubRow.contribution.satoshis / Math.pow(10, 10)
+    totalsRow.votes += pubRow.votes
+    totalsRow.fraction += pubRow.fraction
+    totalsRow.btc += rowBTC
+
+    if (pubRow.contribution.currency === currency) {
+      totalsRow.fiat += parseFloat(pubRow.contribution.fiat || '0')
+    } else {
+      throw new Error('ledgerExportUtil#getTransactionCSVRows does not support mixed currency data (yet)!')
+    }
+
+    return [pub,
+            pubRow.votes,
+            pubRow.fraction,
+            rowBTC,
+            pubRow.contribution.fiat.toFixed(2) + ' ' + pubRow.contribution.currency
+           ].join(',')
+  }))
+
+  if (addTotalRow) {
+    rows.push([
+      totalsRow.label,
+      totalsRow.votes,
+      totalsRow.fraction,
+      totalsRow.btc,
+      totalsRow.fiat.toFixed(2) + ' ' + currency
+    ].join(','))
+  }
+
+  return rows
+}
+
+/**
+ * Generates a contribution breakdown by publisher in an array of CSV rows from an array of transactions
+ * @example
+ * txUtil.getTransactionCSVText(state.transactions)
+ * // 'Publisher,Votes,Fraction,BTC,USD\nchronicle.com,2,0.04081632653061224,0.0000033221,0.20 USD\nwaitbutwhy.com,3,0.061224489795918366,0.0000049832,0.31 USD\narchlinux.org,1,0.02040816326530612,0.0000016611,0.10 USD /.../'
+ *
+ * @param {Object[]} transactions - array of transactions
+ * @param {string[]=} viewingIds - OPTIONAL array/string with one or more viewingIds to filter transactions by (if empty, uses all tx)
+ * @param (boolean=) addTotalRow - OPTIONAL boolean indicating whether to add a TOTALS row (defaults false)
+ **/
+let getTransactionCSVText = function (transactions, viewingIds, addTotalRow) {
+  return getTransactionCSVRows(transactions, viewingIds, addTotalRow).join('\n')
+}
+
+module.exports = {
```

Comment:
for future reference, you can just use es6 object property-value shorthand here. https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand. i.e.,
```
{
  transactionsToCSVDataURL,
  getTransactionCSVText,
 ...
}